### PR TITLE
Add flowgen to create-stub in CLI

### PR DESCRIPTION
**Pull request by goodmind**
- Type of contribution: addition

Other notes:

Related issue #3386 

`flow-typed create-stub ajv --typescript` would use `flowgen`
`flow-typed install` wouldn't use `flowgen` (or you would need to ask user on each untyped dependency to use typescript or not?)

This supports DT, `typings` and `types` fields in `package.json`, or `index.d.ts` if field not found

### CODE REVIEW

**Comment by goodmind** [comment]
@Ville @AndrewSouthpaw

**Comment by AndrewSouthpaw** [comment]
Huh I thought I commented. Can you add a test or some other way of verifying that it works?

**Comment by goodmind** [comment]
@AndrewSouthpaw it depends on what .d.ts file contains. `flowgen` already has ton of tests

**Comment by AndrewSouthpaw** [comment]
Sure, that makes sense. Tests would be great, and also I get that because they're not in place you wouldn't want to make new ones. Would you mind just having a screen gif or at least a demonstrated output so we can know it works?

Also, the build is broken, related to flow typing issues from your code. Those will need to be fixed first.

**Comment by goodmind** [comment]
https://gist.github.com/goodmind/7b6233decb2def0597f1a0852078ab0e

**Comment by goodmind** [comment]
It can of course generate nonsense or even throw errors, this is why it is opt-in

**Comment by goodmind** [comment]
@AndrewSouthpaw build was broken because of old libdefs, well that's no surprise :\

**Comment by AndrewSouthpaw** [comment]
Oh, can you add some notes in the `README` on the new usage? This will be helpful to incoming users.

**Comment by goodmind** [comment]
@AndrewSouthpaw done

File: README.md [hunk 1]
```diff
@@ -109,6 +109,15 @@ a short grace period during a libdef installation, but sometimes it is useful to
 do this update manually. Use this command if you want to download the most
 recent definitions into the cache for yourself.
 
+##### `flow-typed create-stub package-specification [--typescript]`
+
+Creates `any`-typed libdef that you can fill in.
+
+If the `--typescript` flag is specified, the libdef would be converted from `typings` field in package.json, `index.d.ts` file or `@types/package-name` package to flow-typed libdef format.
+
+Note that this option can produce unpredictable output because TypeScript-to-Flow converter is not perfect.
```
Comment (villesau) [inline README.md]: Maybe also mention that manual fixes and careful reviewing of the output is needed, and potentially suggestion about contributing them back to flow-typed?

File: cli/src/lib/npm/npmLibDefs.js [hunk 2]
```diff
@@ -408,8 +408,8 @@ export async function getInstalledNpmLibDefs(
               flowVerMatches == null
                 ? matches[3]
                 : flowVerMatches[3] == null
-                  ? flowVerMatches[2]
-                  : `${flowVerMatches[2]}-${flowVerMatches[4]}`;
+                ? flowVerMatches[2]
+                : `${flowVerMatches[2]}-${flowVerMatches[4]}`;
```
Comment (villesau) [inline cli/src/lib/npm/npmLibDefs.js]: interesting change from prettier O_o I really like the previous formatting more.

File: README.md (hunk 1 above)
Comment (goodmind) [reply README.md]: Isn't this already adressed with `can produce unpredictable output`

Comment (villesau) [reply README.md]: I think it can't be emphasised enough.

Comment (goodmind) [reply README.md]: I updated it

File: cli/src/lib/npm/npmLibDefs.js (hunk 2 above)
Comment (goodmind) [reply cli/src/lib/npm/npmLibDefs.js]: This is just prettier hating nested ternaries

**Comment by goodmind** [comment]
@AndrewSouthpaw @villesau

**Pull request merged**